### PR TITLE
added statistics for dropped flows from flow table

### PR DESCRIPTION
**Pull request by andrekassis**
Warns upon flows being dropped from the flow table due to overflows

### CODE REVIEW

File: flow/probes/ebpf.go [hunk 1]
```diff
@@ -77,6 +76,8 @@ func (p *EBPFProbe) run() {
 	_, flowEBPFChan, _ := p.flowTable.Start()
 	defer p.flowTable.Stop()
 
+	ebpfUpdate := time.Duration(config.GetConfig().GetInt("agent.flow.ebpf_kernel_scan_interval")) * time.Second
```
Comment (safchain) [inline flow/probes/ebpf.go]: can you add an example entry in the default config file etc/skydive.yml.default. We use per probe section, so it should be `agent.flow.ebpf.kernel_scan_interval`

File: flow/probes/ebpf.go [hunk 2]
```diff
@@ -92,6 +93,20 @@ func (p *EBPFProbe) run() {
 		select {
 		case now := <-updateTicker.C:
 			// try to get start monotonic time
+			statsMap := p.module.Map("stats_map")
+			if statsMap == nil {
+				continue
```
Comment (safchain) [inline flow/probes/ebpf.go]: just 

```
if statsMap := p.module.Map("stats_map"); statsMap != nil {
...
}
```

any reason to continue ?

File: flow/probes/ebpf.go [hunk 3]
```diff
@@ -92,6 +93,20 @@ func (p *EBPFProbe) run() {
 		select {
 		case now := <-updateTicker.C:
 			// try to get start monotonic time
```
Comment (safchain) [inline flow/probes/ebpf.go]: this comment is misplaced now

File: flow/probes/ebpf.go [hunk 4]
```diff
@@ -92,6 +93,20 @@ func (p *EBPFProbe) run() {
...
+			statsMap := p.module.Map("stats_map")
+			if statsMap == nil {
+				continue
+			} else {
+				statsKey := uint32(0)
```
Comment (safchain) [inline flow/probes/ebpf.go]: nit: var statsKey uint32

File: flow/table.go [hunk 5]
```diff
@@ -153,6 +154,7 @@ func NewTable(updateHandler *Handler, expireHandler *Handler, nodeTID string, op
 	}
 
 	t.updateVersion = 0
+	t.removedFlows = 0
```
Comment (safchain) [inline flow/table.go]: this line is useless, in fact the upper one too ;)

File: flow/probes/ebpf.go (hunk 1 above)
Comment (andrekassis) [reply flow/probes/ebpf.go]: done

File: flow/probes/ebpf.go (hunk 2 above)
Comment (andrekassis) [reply flow/probes/ebpf.go]: done

File: flow/probes/ebpf.go (hunk 4 above)
Comment (andrekassis) [reply flow/probes/ebpf.go]: done

File: flow/table.go (hunk 5 above)
Comment (andrekassis) [reply flow/table.go]: done for both

File: flow/probes/ebpf.go (hunk 3 above)
Comment (andrekassis) [reply flow/probes/ebpf.go]: done

**Comment by andrekassis** [comment]
run test  skydive-functional-tests-backend-elasticsearch

**Comment by andrekassis** [comment]
run  skydive-functional-tests-backend-elasticsearch

**Comment by andrekassis** [comment]
run skydive-functional-tests-backend-elasticsearch

**Pull request merged**